### ISSUE TITLE
Transport control for enabling playback looping

### ISSUE THREAD

**Issue by jounih** [audacity/audacity#1260]
### User Story
As a new user of Audacity, I would like to have a visible toggle control for playback looping, so it's easy to discover and use.

### Description
Currently in order to loop a selected range in Audacity you have to use the Shift modifier key to show loop play in the UI, or use Shift-Space shortcut for activating it. While this is fine for seasoned Audacity users, it's not very discoverable for new users, and different to almost every other audio software where there is a visible UI looping toggle in the transport controls.

This new designs adds a looping toggle to the transport controls.

<img width="1440" alt="looping off@2x" src="https://user-images.githubusercontent.com/10109867/117793867-93ac6080-b244-11eb-98d3-434a32ee2de2.png">

<img width="1440" alt="loop range selection@2x" src="https://user-images.githubusercontent.com/10109867/122544954-336bc400-d025-11eb-8ae3-11aece8a9370.png">

<img width="1440" alt="audio selection@2x" src="https://user-images.githubusercontent.com/10109867/122544930-2cdd4c80-d025-11eb-9dee-760fcff6a2c8.png">

### Discussion

https://github.com/audacity/audacity/discussions/869

**Comment by petersampsonaudacity** [comment]
Maybe such a loop-play button should be colored green the same as the normal Play button - then it might be more "discoverable" that is is a button for a form of playback

**Comment by Paul-Licameli** [comment]
Are you familiar with all that already exists to define a play region in the time ruler, and the quick-play actions?

Will these features replace those?

@SteveDaulton @DavidBailes

**Comment by SteveDaulton** [comment]
> all that already exists to define a play region in the time ruler, and the quick-play actions?

The Timeline documentation: https://manual.audacityteam.org/man/timeline.html

**Comment by jounih** [comment]
> Maybe such a loop-play button should be colored green the same as the normal Play button - then it might be more "discoverable" that is is a button for a form of playback

The loop button in most DAWs/audio software is actually just a toggle for whether loop playback is enabled or not. The loop button doesn't start playback itself - you still need to hit play. There are some screenshots of other DAWs here - https://github.com/audacity/audacity/discussions/869#discussioncomment-722957

**Comment by gera-gas** [comment]
Design of this feature [here](https://www.figma.com/file/odPgVM0lWc6MdsJWxpgPJl/Audacity?node-id=1657%3A2905)

Some clarifications after the design review:
 * the "quick play" functionality will be replaced on "playback looping".
 * possible simplifications (like in current "quick play" mode) for looping during playback.

**Implementation stages:**
 1. Create and add icon resources.
    1.1. [Methodics](https://wiki.audacityteam.org/wiki/Theming_HowTo)
    1.2. Estimation: 4,6 hrs

 2. Add playback looping button.
     2.1. Methodics: in `ControlToolBar.cpp` file need add looping button by `ControlToolBar::MakeButton` and use `mPause` button like as example for handlers implementation.
    2.2. Estimation: 3,5 hrs

3. Add looping range GUI for Timeline bar.
   3.1. Methodics: Use `QuickPlayIndicatorOverlay` and `QuickPlayRulerOverlay` in `AdornedRulerPanel.cpp` like an example.
   3.2. Estimation: 14,18 hrs

4. Add mouse event handler for looping range GUI.
   4.1. Methodics: Using "quick play" events handlers (`AdornedRulerPanel::QPHandle::Click (const TrackPanelMouseEvent &event, AudacityProject *pProject)`, etc) with looping by Shift key for add playback looping support.
   4.2. Estimation: 12,15 hrs

5. Remove "quick play" preferences like a checkboxes.
    5.1. Note: do it after full "playback looping" support and tests.
    5.2 Estimation: 1.2 hrs

__Final estimation__: 34, 46 hrs

**Comment by SteveDaulton** [comment]
Note that if you intend to replace TimeLine Quick Play, it has several other features in addition to simply looping.
For example, an important feature for creating long loop files is the ability to initiate loop playback from a point close to the end of the loop (so that you don't have to wait for the entire loop to play before you hear if the end and beginning transition correctly). 
Also, the ability to temporarily disable track scrolling during loop playback is very handy when  adjusting the start / end points of the loop.

**Comment by Paul-Licameli** [comment]
I don't think QuickPlayIndicatorOverlay is a useful example.  If you move the mouse in the time ruler without clicking, you can see a white vertical line drawn over the tracks.  That class does this.  The new design does not require this.

I think the changes of behavior call all be implemented inside AdornedRulerPanel::OnPaint() and overlays need not be part of it.

**Comment by petersampsonaudacity** [comment]
The other key thing that Timeline Quickplay is uses for apart from looping is to easily identify points of interest in a track, i'e simple playback with no looping.

In other audio and video apps (Reaper as a good example) one would use scrubbing for for this - but audacity's scrubbing has an odd GUI that provides for a poor UX experience - and thus users (me included) will use Timeline Quickplay instead.

**Comment by jounih** [comment]
I don't think we want to deprecate Quickplay but it shouldn't be the default action from selecting a time range, it looks like looping, happens in an area where most users expect setting a loop region to be the default action -  and confuses most new users because because of this.

I think we can still retain Quickplay but perhaps with a modifier key + region selection, rather than by default.

And agree, we need to improve scrubbing also so people can use that!

**Comment by jounih** [comment]
> Note that if you intend to replace TimeLine Quick Play, it has several other features in addition to simply looping.
> For example, an important feature for creating long loop files is the ability to initiate loop playback from a point close to the end of the loop (so that you don't have to wait for the entire loop to play before you hear if the end and beginning transition correctly).

You should be able to do this with the new playback looping also - just set a region you want to loop, hit play, and then click somewhere else on the timeline to jump to that time (while playing). Looping would still enabled as long as you don't toggle it off with the loop button. This is how it works in most other DAWs
